src/unistd.rs

```diff
@@ -2729,3 +2751,42 @@ impl Group {
         })
     }
 }
+
+
+#[cfg(any(target_os = "linux", target_os = "android"))]
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use std::{fs, thread, io};
+
+    /// Tests setting the filesystem UID with `setfsuid`.
+    #[test]
+    fn test_setfsuid() {
+        use std::os::unix::fs::PermissionsExt;
+
+        // get the UID of the "nobody" user
```

Comment:
Per the man page, `setfsuid` requires the `CAP_SETUID` capability.  You should check for that here, like so:
```rust
require_capability!(CAP_SETUID);
```

Reply:
Added it.